File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        
        // int[] -> String[]
        String[] strArr = new String[numbers.length];
        for (int i = 0; i < numbers.length; i++) {
            // String.valueOf() 메서드로 숫자를 문자열로 변환
            strArr[i] = String.valueOf(numbers[i]);
        }
        
        // 내림차순 정렬
        Arrays.sort(strArr, new Comparator<String>() {
            @Override
            public int compare(String a, String b) {
                return (b+a).compareTo(a+b);
            }
        });
        
        // 0이 중복될 경우(ex. {0, 0, 0}) -> 배열의 첫번째 값이 0이면 0을 출력
        if (strArr[0].equals("0")) return "0";
        
        // 0이 아니면 문자열을 answer에 더한다
        for (String str : strArr) {
            answer += str;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    # string형 list로 변환
    str_numbers = list(map(str, numbers))
    # 문자열 타입으로 정렬
    str_numbers.sort(key = lambda x: x*3, reverse=True)
    # '000', '00' 등을 '0'으로 반환하기 위해 join하여 int형 -> 문자열
    return str(int(''.join(str_numbers)))

print(solution([6, 10, 2]))          
print(solution([3, 30, 34, 5, 9]))    
print(solution([0,0,0,0]))